mrp_unbuild_rebuild_variant/models/mrp_unbuild_rebuild_variant.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import _, api, exceptions, fields, models
+
+
+class MrpUnbuildRebuildVariant(models.Model):
+
+    _name = "mrp.unbuild.rebuild.variant"
+
+    name = fields.Char(
+        default=lambda self: self.env["ir.sequence"].next_by_code(self._name),
+        readonly=True,
+    )
+    state = fields.Selection(
+        [("draft", "Draft"), ("done", "Done")],
+        required=True,
+        default="draft",
+        readonly=True,
+    )
+    # TODO: WHAT SHOULD BE COMPUTED HERE
+    unbuild_product_id = fields.Many2one("product.product")
+    unbuild_bom_id = fields.Many2one("mrp.bom", compute="_compute_unbuild_product_id")
+    unbuild_location_id = fields.Many2one("stock.location")
+    # if unbuild product is tracked:
+    unbuild_lot_id = fields.Many2one(
+        "stock.production.lot", domain="[('product_id', '=', unbuild_product_id)]"
+    )
+    # if unbuild product's bom includes tracked components:
+    unbuild_original_mo_id = fields.Many2one("mrp.production")
+    # if unbuild product's bom does not include tracked components:
+    quantity = fields.Float()
+
+    rebuild_product_template = fields.Many2one(
+        "product.template", compute="_compute_valid_template"
+    )
+
+    # TODO: Domain for same product.template
+    rebuild_product_id = fields.Many2one(
+        "product.product",
+        # FIXME: domain="[
```

Comment:
can you leave more human-friendly comment about the over left

Reply:
We need to define a domain based on a computed field here. Something like this (untested):
```
allowed_rebuild_product_ids = fields.Many2many("product.product", compute="_compute_allowed_rebuild_product_ids")
rebuild_product_id = fields.Many2one("product.product", domain="[('id', 'in', allowed_rebuild_product_ids), ('id', '!=', unbuild_product_id)]", required=True)

@api.depends("unbuild_product_id")
def _compute_allowed_rebuild_product_ids(self):
    for rec in self:
        rec.allowed_rebuild_product_ids = rec.unbuild_product_id.mapped("product_tmpl_id.product_variant_ids").ids
```